File: n1/1.py
#Функция для нахождения простого числа
def smpl(n):
    if n < 2:
        return 0
    for i in range(2, int(n**0.5) + 1):
        if n % i == 0:
            return 0
    return 1

N = int(input())

#Поиск ближайшего простого числа, меньшего или равного N
smaller = N
while not smpl(smaller):
    smaller -= 1

#Поиск ближайшего простого числа большего N
larger = N
while not smpl(larger):
    larger += 1

#Сравнивнение расстояния
if (N - smaller) <= (larger - N):
    print(N - smaller)
else:
    print(larger - N)
python code

File: n1/1.php
<?php
// Функция для проверки, является ли число простым
function smpl($n) {
    if ($n < 2) return false;
    for ($i = 2; $i <= sqrt($n); $i++) {
        if ($n % $i == 0) return false;
    }
    return true;
}
echo "Введите число для определения разности между ним и ближайшим простым: ";
$N = intval(trim(fgets(STDIN)));

// Поиск ближайшего простого числа <= N
$smaller = $N;
while (!smpl($smaller)) $smaller--;

// Поиск ближайшего простого числа >= N
$larger = $N;
while (!smpl($larger)) $larger++;

// Сравнение расстояний и вывод результата
echo "Разность: " . min($N - $smaller, $larger - $N) . PHP_EOL;
?>
